src/vendor/azure-uamqp-c/src/link.c

```diff
@@ -278,6 +280,15 @@ static int send_attach(LINK_INSTANCE* link, const char* name, handle handle, rol
         {
             (void)attach_set_properties(attach, link->attach_properties);
         }
+        
+        if (link->desired_capabilities != NULL)
+        {
+            if(attach_set_desired_capabilities(attach, link->desired_capabilities) != 0)
+            {
+                LogError("Cannot set attach desired capabilities");
+                result = __FAILURE__;
+            }
+        }
```

Comment:
I see we're setting the desired capabilities here, but we're not extracting the offered capabilities from the received attach frame?

Reply:
Yes, it seems that to enable the runtime-metric feature of eventhubs, we just need to set the desired_capabilities of the link - you can see how i create the properties and set it on the receiver [here](https://github.com/Azure/azure-sdk-for-python/blob/9928d9cd31b4e880246fae8f5225c73061c8e607/sdk/eventhub/azure-eventhubs/azure/eventhub/aio/consumer_async.py#L135), this can work.
I don't use `get_desired_capabilities`.

Reply:
If we don't need to get offered_capabilities - why add the extra functions to support it?

Reply:
I added them because the `offered_capabilities` is part of the protocol and is being missed in our implementation. I want to implement the protocol.😃

one thing I'm not doing good enough is that I didn't extract the offer_capabilities from the attach frame which makes the `desired_properties` seems useless...

maybe you're right.. I should skip it for now

Reply:
you're right, previously I thought this attribute would be useful, but right now it seems not.
I'll remove it.